pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,871 @@
+						ClaimName: pvc.Name,
+					},
+				},
+			})
+		}
+		ctrl.addVolumeEnvironmentVariables(&podManifest.Spec.Containers[0], pvc, i, mountPoint)
+	}
+
+	// Add token and certs ENV variables
+	podManifest.Spec.Containers[0].Env = append(podManifest.Spec.Containers[0].Env, corev1.EnvVar{
+		Name:  "CERT_FILE",
+		Value: "/cert/tls.crt",
+	}, corev1.EnvVar{
+		Name:  "KEY_FILE",
+		Value: "/cert/tls.key",
+	}, corev1.EnvVar{
+		Name:  "TOKEN_FILE",
+		Value: "/token/token",
+	}, corev1.EnvVar{
+		Name:  "DEADLINE",
+		Value: currentTime().Add(time.Hour * deadline).Format(time.RFC3339),
+	})
+
+	secretName := fmt.Sprintf("secret-%s", rand.String(10))
+	podManifest.Spec.Volumes = append(podManifest.Spec.Volumes, corev1.Volume{
+		Name: certificates,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	}, corev1.Volume{
+		Name: vmExport.Spec.TokenSecretRef,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: vmExport.Spec.TokenSecretRef,
+			},
+		},
+	})
+
+	podManifest.Spec.Containers[0].VolumeMounts = append(podManifest.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+		Name:      certificates,
+		MountPath: "/cert",
+	}, corev1.VolumeMount{
+		Name:      vmExport.Spec.TokenSecretRef,
+		MountPath: "/token",
+	})
+	return podManifest
+}
+
+func (ctrl *VMExportController) addVolumeEnvironmentVariables(exportContainer *corev1.Container, pvc *corev1.PersistentVolumeClaim, index int, mountPoint string) {
+	exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+		Name:  fmt.Sprintf("VOLUME%d_EXPORT_PATH", index),
+		Value: mountPoint,
+	})
+	if pvc.Spec.VolumeMode != nil && *pvc.Spec.VolumeMode == corev1.PersistentVolumeBlock {
+		exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+			Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_URI", index),
+			Value: rawURI(pvc),
+		}, corev1.EnvVar{
+			Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_GZIP_URI", index),
+			Value: rawGzipURI(pvc),
+		})
+	} else {
+		if ctrl.isKubevirtContentType(pvc) {
+			exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_URI", index),
+				Value: rawURI(pvc),
+			}, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_GZIP_URI", index),
+				Value: rawGzipURI(pvc),
+			})
+		} else {
+			exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_ARCHIVE_URI", index),
+				Value: archiveURI(pvc),
+			}, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_DIR_URI", index),
+				Value: dirURI(pvc),
+			})
+		}
+	}
+}
+
+func (ctrl *VMExportController) isKubevirtContentType(pvc *corev1.PersistentVolumeClaim) bool {
+	ann := pvc.GetAnnotations()
+	if ann == nil {
+		return false
+	}
+	return ann[annContentType] == string(cdiv1.DataVolumeKubeVirt) || ann[annContentType] == ""
```

Comment:
Kind of weird logic here:

If no annotations, return false
If content type annotation missing but other annotations exist, return true
If content type annotation set to "" or DataVolumeKubeVirt return true

Reply:
Yeah not sure why I am returning false on no annotations, fixing that.